### ISSUE TITLE
FirstWhere rule wrongly triggers on `filter().first(where:)`

### ISSUE THREAD

**Issue by ldiqual** [realm/SwiftLint#1426]
The following code should not trigger the error:

```swift
let pickupWaypoint = tripMatch.waypoints(includeCancelled: false)
    .filter { $0.account == currentAccount }
    .first(where: { $0.action == .pickup })
```

![image](https://cloud.githubusercontent.com/assets/941453/24898334/6a0c127e-1e51-11e7-9f9e-587532ae74c2.png)

**Comment by jpsim** [comment]
What makes you think that's a false positive? It's a legitimate violation as far as I can tell since you can rewrite to this, which is equivalent but more efficient:

```swift
let pickupWaypoint = tripMatch.waypoints(includeCancelled: false)
    .first(where: { $0.account == currentAccount && $0.action == .pickup })
```

**Comment by RishabhTayal** [comment]
I am getting this false positive in my project.

`if let dateStr = queryItems?.first(where: { $0.name == "date" })?.value {`

![screenshot 2017-04-14 16 31 38](https://cloud.githubusercontent.com/assets/2100166/25056657/e558e458-212f-11e7-8416-84795f97290d.png)

**Comment by jpsim** [comment]
@RishabhTayal what you've shared _is_ a false positive, whereas the example shared by @ldiqual is not.

However, I can't reproduce your issue with the latest version of SwiftLint (0.18.1):

```
$ swiftlint version
0.18.1
$ cat first_where.swift
struct Item {
  let name: String
  let value: String?
}
let queryItems: [Item]? = [Item(name: "age", value: "0"), Item(name: "date", value: "1")]

if let dateStr = queryItems?.first(where: { $0.name == "date" })?.value {
  print(dateStr)
}

if let thisViolates = queryItems?.filter({ $0.value != nil }).first(where: { $0.name == "date" })?.value {
  print(thisViolates)
}
$ cat .swiftlint.yml
opt_in_rules:
  - first_where
$ swiftlint lint --path first_where.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at path first_where.swift
Linting 'first_where.swift' (1/1)
/path/to/first_where.swift:11:23: warning: First Where Violation: Prefer using `.first(where:)` over `.filter { }.first` in collections. (first_where)
Done linting! Found 1 violation, 0 serious in 1 file.
```

Are you sure you're using SwiftLint 0.18.1? If so, please file a new issue with complete steps to reproduce. Thanks!

**Comment by RishabhTayal** [comment]
@jpsim I believe I was using 0.16.1 version. I just upgraded to 0.18.1. I will see if I can reproduce now.

Thanks.
